frontend/src/components/Board/Board.tsx

```diff
@@ -1,7 +1,339 @@
+function TaskCard(props: { item: ICard }) {
+  const item = props.item;
+  // console.log(item)
+  return (
+    <>
+      <Card className='m-1' sx={{ maxWidth: 345 }}>
+        <CardContent>
+          <Typography gutterBottom variant='h6' component='div'>
+            {item.title}
+          </Typography>
+          <Typography variant='body2' color='text.secondary'>
+            {item.description}
+          </Typography>
+        </CardContent>
+        <CardActions className='flex flex-col'>
+          <Button className='justify-self-end' size='small'>
+            Edit
+          </Button>
+        </CardActions>
+      </Card>
+    </>
+  );
+}
+
+function TaskColumn(props: IColumnProp) {
+  const column = props.column;
+  const items = props.column.cards
+    .filter((i: ICard) => i.list_id == column.id)
+    .sort((c) => c.index);
+  // console.log("Column: ", {column}, {items})
+  return (
+    <>
+      <div className='flex flex-col h-min-full w-full min-w-[200px] mx-2 px-2 bg-gray-500 rounded-md pt-2'>
+        <div className='w-full bg-gray-100 text-black font-bold text-xl rounded-md text-center border-2 border-gray-700'>
+          {column.name}
+        </div>
+        <div className='w-full h-full overflow-y-auto'>
+          <Droppable droppableId={column.id}>
+            {(provided) => (
+              <div
+                {...provided.droppableProps}
+                ref={provided.innerRef}
+                className='min-h-[95%]'
+              >
+                {items.map((item: ICard, index: number) => (
+                  <Draggable key={item.id} draggableId={item.id} index={index}>
+                    {(provided) => (
+                      <div
+                        ref={provided.innerRef}
+                        {...provided.draggableProps}
+                        {...provided.dragHandleProps}
+                      >
+                        <TaskCard item={item} />
+                      </div>
+                    )}
+                  </Draggable>
+                ))}
+                {provided.placeholder}
+              </div>
+            )}
+          </Droppable>
+        </div>
+      </div>
+    </>
+  );
+}
+
+function TaskBoard() {
+  const [myColumns, setMyColumns] = useState(columns);
+
+  // Handles moving cards
+  function onDragEnd(result: any) {
+    // console.log("drag ended ", result);
+    const columnSource: IColumn | undefined = columns?.find(
+      (c) => c.id == result?.source?.droppableId
+    );
+    const columnDestination: IColumn | undefined = columns?.find(
+      (c) => c.id == result?.destination?.droppableId
+    );
+    const cardToMove: ICard | undefined = columnSource?.cards?.find(
+      (c) => c.id == result?.draggableId
+    );
+    if (cardToMove && columnSource && columnDestination) {
+      cardToMove.list_id = result?.destination?.droppableId;
+      // If moving cards in the same column
+      if (result?.source?.droppableId == result?.destination?.droppableId) {
```

Comment:
I'm guessing the order of splicing columnSource or columnDestination matters? (It's the only difference I can tell between the if and else)

Reply:
Thats my 5am brain thinking, but its what we need for the difference between switching cards between columns and in the same column.